flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -1090,14 +1147,17 @@ struct EmboxOpConversion : public EmboxCommonConversion<fir::EmboxOp> {
                   mlir::ConversionPatternRewriter &rewriter) const override {
     // There should be no dims on this embox op
     assert(!embox.getShape());
+    int rank = 0;
+    int dropFront = 1;
```

Comment:
Why were these made variables? I don't see them changing below.

Reply:
They don't but my changes used `rank` in several places and I find it safer for it to be defined in a single location (I should have used `const` though).
However, I moved getStoreOrInsertField to be called by `consDescriptorPrefix` instead, so I rolled that back.